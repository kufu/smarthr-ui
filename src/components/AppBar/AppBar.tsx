import * as React from 'react'
import styled, { css } from 'styled-components'

import { ComponentProps, Size, StyledProperties } from '../../types/props'
import { useTheme } from '../../hooks/useTheme'

export const AppBar: React.FC<ComponentProps> = ({ ...props }) => {
  const theme = useTheme()
  return <Wrapper theme={theme} {...props} />
}

const getSpaceSize = (size: Size): 'XS' | 'S' | 'M' => {
  const spaceMap: any = {
    s: 'XS',
    m: 'S',
    l: 'M',
  }
  return spaceMap[size]
}
const Wrapper = styled.div<StyledProperties>`
  ${({ spSize = 'm', pcSize = 'm', tabletSize = 'm', theme }) => {
    const { size, palette } = theme
    const [pcPadding, tabletPadding, spPadding] = [pcSize, tabletSize, spSize]
      .map(getSpaceSize)
      .map(spaceSize => `0 ${size.pxToRem(size.space[spaceSize])}`)

    return css`
      padding: ${pcPadding};
      background: ${palette.MAIN};

      @media screen and (max-width: ${size.mediaQuery.TABLET}px) {
        padding: ${tabletPadding};
      }

      @media screen and (max-width: ${size.mediaQuery.SP}px) {
        padding: ${spPadding};
      }
    `
  }}
`
