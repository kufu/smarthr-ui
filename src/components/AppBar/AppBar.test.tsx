import { mount } from 'enzyme'
import React from 'react'
import ReactTestRenderer from 'react-test-renderer'

import { AppBar } from './AppBar'

describe('AppBar', () => {
  it('should be match snapshot', () => {
    const component = mount(<AppBar />)
    expect(component).toMatchSnapshot()
  })

  it('should be able to render without crashing', () => {
    ReactTestRenderer.create(<AppBar />)
  })
})
