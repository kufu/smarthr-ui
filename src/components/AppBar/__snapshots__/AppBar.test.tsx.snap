// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should be match snapshot 1`] = `
<Component>
  <AppBarComponent
    theme={
      Object {
        "frame": Object {
          "border": Object {
            "default": "1px solid #d6d6d6",
            "lineStyle": "solid",
            "lineWidth": "1px",
            "radius": Object {
              "l": "8px",
              "m": "6px",
              "s": "3px",
            },
          },
        },
        "interaction": Object {
          "hover": Object {
            "animation": ".3s ease-out",
            "animationDuration": ".3s",
            "animationTiming": "ease-out",
            "feedbackOpacity": ".7",
          },
        },
        "palette": Object {
          "Background": "#f5f6fa",
          "Border": "#d6d6d6",
          "Column": "#f9f9f9",
          "Danger": "#ef475b",
          "Hanica_Green": "#57d0d5",
          "Main": "#00a5ab",
          "Overlay": "rgba(0,0,0,0.15)",
          "Scrim": "rgba(0,0,0,0.5)",
          "TextBlack": "#333",
          "TextGrey": "#767676",
          "Warning": "#ff8800",
          "hoverColor": [Function],
        },
        "size": Object {
          "font": Object {
            "grande": 16,
            "short": 13,
            "tall": 14,
            "tasting": 12,
            "trenta": 22,
            "venti": 20,
          },
          "mediaQuery": Object {
            "sp": 599,
            "tablet": 959,
          },
          "pxToRem": [Function],
          "space": Object {
            "l": 40,
            "m": 32,
            "s": 24,
            "xl": 48,
            "xs": 16,
            "xxl": 56,
            "xxs": 8,
          },
        },
      }
    }
  >
    <styled.div
      theme={
        Object {
          "frame": Object {
            "border": Object {
              "default": "1px solid #d6d6d6",
              "lineStyle": "solid",
              "lineWidth": "1px",
              "radius": Object {
                "l": "8px",
                "m": "6px",
                "s": "3px",
              },
            },
          },
          "interaction": Object {
            "hover": Object {
              "animation": ".3s ease-out",
              "animationDuration": ".3s",
              "animationTiming": "ease-out",
              "feedbackOpacity": ".7",
            },
          },
          "palette": Object {
            "Background": "#f5f6fa",
            "Border": "#d6d6d6",
            "Column": "#f9f9f9",
            "Danger": "#ef475b",
            "Hanica_Green": "#57d0d5",
            "Main": "#00a5ab",
            "Overlay": "rgba(0,0,0,0.15)",
            "Scrim": "rgba(0,0,0,0.5)",
            "TextBlack": "#333",
            "TextGrey": "#767676",
            "Warning": "#ff8800",
            "hoverColor": [Function],
          },
          "size": Object {
            "font": Object {
              "grande": 16,
              "short": 13,
              "tall": 14,
              "tasting": 12,
              "trenta": 22,
              "venti": 20,
            },
            "mediaQuery": Object {
              "sp": 599,
              "tablet": 959,
            },
            "pxToRem": [Function],
            "space": Object {
              "l": 40,
              "m": 32,
              "s": 24,
              "xl": 48,
              "xs": 16,
              "xxl": 56,
              "xxs": 8,
            },
          },
        }
      }
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": false,
              "lastClassName": "ineNls",
              "rules": Array [
                "
  ",
                [Function],
                "
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        theme={
          Object {
            "frame": Object {
              "border": Object {
                "default": "1px solid #d6d6d6",
                "lineStyle": "solid",
                "lineWidth": "1px",
                "radius": Object {
                  "l": "8px",
                  "m": "6px",
                  "s": "3px",
                },
              },
            },
            "interaction": Object {
              "hover": Object {
                "animation": ".3s ease-out",
                "animationDuration": ".3s",
                "animationTiming": "ease-out",
                "feedbackOpacity": ".7",
              },
            },
            "palette": Object {
              "Background": "#f5f6fa",
              "Border": "#d6d6d6",
              "Column": "#f9f9f9",
              "Danger": "#ef475b",
              "Hanica_Green": "#57d0d5",
              "Main": "#00a5ab",
              "Overlay": "rgba(0,0,0,0.15)",
              "Scrim": "rgba(0,0,0,0.5)",
              "TextBlack": "#333",
              "TextGrey": "#767676",
              "Warning": "#ff8800",
              "hoverColor": [Function],
            },
            "size": Object {
              "font": Object {
                "grande": 16,
                "short": 13,
                "tall": 14,
                "tasting": 12,
                "trenta": 22,
                "venti": 20,
              },
              "mediaQuery": Object {
                "sp": 599,
                "tablet": 959,
              },
              "pxToRem": [Function],
              "space": Object {
                "l": 40,
                "m": 32,
                "s": 24,
                "xl": 48,
                "xs": 16,
                "xxl": 56,
                "xxs": 8,
              },
            },
          }
        }
      >
        <div
          className="sc-bdVaJa ineNls"
        />
      </StyledComponent>
    </styled.div>
  </AppBarComponent>
</Component>
`;
